crates/sncast/src/response/print.rs

```diff
@@ -0,0 +1,242 @@
+use anyhow::Result;
+use serde_json::Value;
+use starknet::core::types::FieldElement;
+use std::{collections::HashMap, fmt::Display, str::FromStr};
+
+use serde::{Serialize, Serializer};
+
+use crate::NumbersFormat;
+
+use super::structs::CommandResponse;
+
+pub enum OutputFormat {
+    Json,
+    Human,
+}
+
+impl OutputFormat {
+    #[must_use]
+    pub fn from_flag(json: bool) -> Self {
+        if json {
+            OutputFormat::Json
+        } else {
+            OutputFormat::Human
+        }
+    }
+}
+
+/// Constrained subset of `serde::json`
+#[derive(PartialEq, Eq, Debug)]
+pub enum OutputValue {
+    String(String),
+    Array(Vec<OutputValue>),
+}
+
+type OutputData = Vec<(String, OutputValue)>;
+
+impl Serialize for OutputValue {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match &self {
+            OutputValue::String(s) => serializer.serialize_str(s),
+            OutputValue::Array(arr) => arr.serialize::<S>(serializer),
+        }
+    }
+}
+
+impl Display for OutputValue {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match &self {
+            OutputValue::String(s) => s.fmt(fmt),
+            OutputValue::Array(arr) => {
+                let arr_as_string = arr
+                    .iter()
+                    .map(ToString::to_string)
+                    .collect::<Vec<String>>()
+                    .join(", ");
+                write!(fmt, "[{arr_as_string}]")
+            }
+        }
+    }
+}
+
+pub fn print_command_result<T: CommandResponse>(
+    command: &str,
+    result: &mut Result<T>,
+    numbers_format: NumbersFormat,
+    output_format: &OutputFormat,
+) -> Result<()> {
+    let mut output: OutputData = vec![];
+    output.push((
+        String::from("command"),
+        OutputValue::String(command.to_string()),
+    ));
+    output.extend(result_as_output_data(result));
+    let formatted_output = output
+        .into_iter()
+        .map(|(k, v)| (k, apply_numbers_formatting(v, numbers_format)))
+        .collect();
+
+    for val in pretty_output(formatted_output, output_format)? {
+        match result {
+            Ok(_) => println!("{val}"),
+            Err(_) => eprintln!("{val}"),
+        }
+    }
+    Ok(())
+}
+
+fn pretty_output(output: OutputData, output_format: &OutputFormat) -> Result<Vec<String>> {
+    match output_format {
+        OutputFormat::Json => {
+            let json_output: HashMap<String, OutputValue> = output.into_iter().collect();
+            let pretty_string = serde_json::to_string(&json_output)?;
+            Ok(vec![pretty_string])
+        }
```

Comment:
```suggestion
            let json_output = serde_json::to_string(&json_output)?;
            Ok(vec![json_output])
        }
```

since `serde_json::to_string` is used instead of `serde_json::to_string_pretty`